### PR TITLE
Update image dimensions to benefit from new srcset generation

### PR DESCRIPTION
**Pull request by sebastianbenz**
This makes use of the new AMP Optimizer srcset generation feature.
Templates need to be updated to use the mobile dimensions for width and
height.

### CODE REVIEW

**Comment by sebastianbenz** [comment]
First experiment. Not really feasible as this triggers the image optimization on every page load.

File: platform/lib/utils/imageOptimizer.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * limitations under the License.
+ */
+
+const path = require('path');
+
+const Image = require('@11ty/eleventy-img');
+
+const DEFAULT_IMG_FORMAT = 'jpeg';
+
+const baseDir = path.join(__dirname, '../../..');
```
Comment (matthiasrohmer) [inline platform/lib/utils/imageOptimizer.js:23]: I'd suggest using the project helper to build the paths:

```js
const { project } = require('@lib/utils');

const baseDir = project.absolute('/');
```

Or for this specific case there's also:

```js
const baseDir = project.paths.ROOT;
```

File: platform/lib/utils/imageOptimizer.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+const imageBaseDir = path.join(baseDir, 'pages');
+const outputDir = path.join(baseDir, 'dist/static/img');
+
+const globalOpts = {
+  urlPath: '/static/img/',
```
Comment (matthiasrohmer) [inline platform/lib/utils/imageOptimizer.js:28]: Maybe add `formats` to make it obvious it's a supported option?

File: platform/lib/utils/imageOptimizer.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+};
+
+/**
+ * Resizes the image specified at the given src URL to the given width. Images
+ * specified via an absolute URL will be downloaded and cached locally.
+ *
+ * @param {string} src - the original img's src URL
+ * @param {number} width  - the target width
+ */
+const imageOptimizer = async (src, width) => {
```
Comment (matthiasrohmer) [inline platform/lib/utils/imageOptimizer.js:39]: Why not `async function imageOptimizer(src, width)`?

File: platform/lib/utils/imageOptimizer.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      widths: [width],
+    });
+    if (!isAbsoluteUrl(src)) {
+      src = path.join(imageBaseDir, src);
+    }
+    // Resizes, compresses the image (and download if needed).
+    const stats = await Image(src, opts); // eslint-disable-line
+    return stats[format][0].url;
+  } catch (e) {
+    console.log(`Could not optimize image (${src}, ${width}w):`, e);
```
Comment (matthiasrohmer) [inline platform/lib/utils/imageOptimizer.js:53]: Use the log helper for scoping and consistency:

```js
const log = require('@lib/utils/log')('Image Optimizer');
log.warn(`Could not optimize image (${src}, ${width}w):`, e);
```

**Comment by sebastianbenz** [comment]
@matthiasrohmer PTAL - I've removed the image optimization parts and only left the template improvements

**Pull request merged**